src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```
@@ -114,8 +114,80 @@ def setup(self, newparams):
+
+                # channel ID
+                if chan == 0:
+                    channel_id = ''
+                else:
+                    channel_id = f'_{chan}'
+
+                for n in range(nspots):
+                    # read radii
+                    item0 = 'spotrad' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotrad[chan, n] = value
+                    # read latitudes
+                    item0 = 'spotlat' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotlat[chan, n] = value
+                    # read longitudes
+                    item0 = 'spotlon' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotlon[chan, n] = value
+                # read contrasts (same for all spots)
+                item0 = 'spotcon0'
+                if self.parameters.dict[item0][1] == 'free':
+                    item0 += channel_id
+                value = self.parameters.dict[item0][0]
+                spotcon[chan] = value
+                # read stellar inclination
+                item0 = 'spotstari'
+                if self.parameters.dict[item0][1] == 'free':
+                    item0 += channel_id
+                value = self.parameters.dict[item0][0]
+                starinc[chan] = value
+                # read number of points
+                starres = self.parameters.dict['spotnpts'][0]
+                # read stellar rotation (if provided)
+                if 'spotrot' in self.parameters.dict.keys():
+                    starrot[chan] = self.parameters.dict['spotrot'][0]
```

In the `setup` function we're pre-compiling the code to be evaluated analytically by theano/pymc3, so we need to use the theano tensor variables in the object `self.model` (using `getattr` since it is an object instead of a dictionary) instead of using the dictionary `self.parameters.dict` (which only contains the initial `float` guesses and would result in the `pymc3` fits silently failing since the star spot parameters were never used in the code)

---

But in the `update` function down below, we're using the updated "best-fit" values after the fitting process which are stored in the `self.parameters.dict` dictionary, so your code in the `update` function is fine.

---

not gonna lie, a little lost on this one. For some reason I no longer can successfully fit a spot-free light curve with starry so it's hard to test the implementation of spots.